app/controllers/finance/financial_transactions_controller.rb

```diff
@@ -51,6 +51,14 @@ def create
   end
 
   def new_collection
+    @ordergroups = {}
+    Ordergroup.undeleted.order(:name).map do |ordergroup|
+      obj = { name: ordergroup.name }
+      Ordergroup.custom_fields.each do |field|
+        obj[field[:name]] = ordergroup.settings.custom_fields[field[:name]]
+      end
+      @ordergroups[ordergroup.id] = obj
+    end
```

Comment:
this code isn't really very easy to read.  i would suggest at least some comments on what is going on, or perhaps putting all the `custom_fields` stuff in a new method, or even in the model (or a helper).

and perhaps nit picking, but the variable names here are hard to follow.  eg, @ordergroups is not a collection of `ordergroups` but rather the `@ordergroups_custom_fields` (yes a mouthful), and `obj` is  not a great name, and a better name would make it clearer what `@ordergroups` contains.

Reply:
i'm not a big fan of comments. if they are needed the code is not well written. i'l rework it according to your suggestions.

Reply:
for sure - clear code is the best.  but sometimes, a small hint in a comment is helpful.  i agree that comments aren't the best because the worst part is they can become out of date and misleading!